File: web/src/types/BlockParserState.ts
import BlockRule from "./BlockRule";
import type FootnoteReference from "./FootnoteReference";
import type LinkReference from "./LinkReference";
import type MarkdownNode from "./MarkdownNode";

export default interface BlockParserState {
	rules: BlockRule[];
	rulesMap: Map<string, BlockRule>;

	src: string;
	i: number;
	line: number;
	lineStart: number;
	indent: number;
	spaces: string;
	openNodes: MarkdownNode[];
	isEscaped: boolean;
	maybeContinue: boolean;
	hasBlankLine: boolean;
	refs: Record<string, LinkReference>;
	footnotes: Record<string, FootnoteReference>;
}

File: zig/src/types/BlockParserState.zig
const std = @import("std");

pub const BlockParserState = struct {
    allocator: std.mem.Allocator,
    rules: []const *const BlockRule,
    rulesMap: std.StringHashMap(*const BlockRule),
    src: []const u8,
    i: usize,
    line: i32,
    lineStart: usize,
    indent: i32,
    spaces: []const u8,
    isEscaped: bool,
    openNodes: std.ArrayList(*MarkdownNode),
    maybeContinue: bool,
    hasBlankLine: bool,
    refs: std.StringHashMap(LinkReference),
    footnotes: std.StringHashMap(FootnoteReference),
};

const BlockRule = @import("BlockRule.zig").BlockRule;
const MarkdownNode = @import("MarkdownNode.zig").MarkdownNode;
const LinkReference = @import("LinkReference.zig").LinkReference;
const FootnoteReference = @import("FootnoteReference.zig").FootnoteReference;
